src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,77 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.ValidationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int INVALID_AGE = 16;
+    private static final int VALID_AGE = 18;
+    private static final long VALID_ID = 1234567890L;
+    private static final String INVALID_PASSWORD = null;
+    private static final String VALID_PASSWORD = "qwerty";
+    private static final String VALID_LOGIN = "nameEx";
+    private static final String INVALID_LOGIN = "name";
+
+    private final User user = new User();
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @BeforeEach
+    void setUp() {
+        user.setAge(VALID_AGE);
+        user.setId(VALID_ID);
+        user.setPassword(VALID_PASSWORD);
+        user.setLogin(VALID_LOGIN);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    public void register_AddToStorage_Ok() {
+        User registeredUser = registrationService.register(user);
+        assertEquals(user, registeredUser, "The registered user should match the valid user");
+    }
+
+    @Test
+    public void register_invalidAge_notOk() {
+        user.setAge(INVALID_AGE);
+        ValidationException exception = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Validation error",
+                exception.getMessage(), "Incorrect validation error message");
+    }
+
+    @Test
+    public void register_invalidLogin_notOk() {
+        user.setLogin(INVALID_LOGIN);
+        ValidationException exception = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Validation error",
+                exception.getMessage(), "Incorrect validation error message");
+    }
+
+    @Test
+    public void register_invalidPassword_notOk() {
+        user.setPassword(INVALID_PASSWORD);
+        ValidationException exception = assertThrows(ValidationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Validation error",
+                exception.getMessage(), "Incorrect validation error message");
+    }
+
+    @Test
+    public void register_sameLogin_notOk() {
+        assertFalse(Storage.people.contains(user),
+                "User with the same login should not exist");
+    }
```

Comment:
Right now you are just asserting that there is no user with such login in the storage, which is expected because there is no logic in putting it there, but it should test the the `register()` behavior when there is already a user in the storage, and we try to register another one with the same login
order of actions for this test: 
* put the user to the storage(without using the `register()` method, because it is under the test, see common mistakes for details)
* try to register a user with the same login
* assert the resulting exception and it's message